lib/src/booleans/boolean_touches.dart

```diff
@@ -0,0 +1,590 @@
+import '../../helpers.dart';
+import 'boolean_point_in_polygon.dart';
+import 'boolean_point_on_line.dart';
+
+/// Boolean-touches [true] if none of the [Point]s common to both geometries
+/// intersect the interiors of both geometries.
+/// example
+/// ```dart
+/// var line = LineString(coordinates;[Positin.of([1, 1]), Positin.of([1, 2]), Positin.of([1, 3]), Positin.of([1, 4])]);
+/// var point = Point(coordinates: Positon.of([1, 1]));
+/// booleanTouches(point, line);
+/// //=true
+/// ```
+bool booleanTouches(GeoJSONObject feature1, GeoJSONObject feature2) {
+  var geom1 = feature1 is Feature ? feature1.geometry : feature1;
+  var geom2 = feature2 is Feature ? feature2.geometry : feature2;
+
+  if (geom1 is Point) {
```

Comment:
this is a super long implementation, was it like that in turf js? (if yes, just resolve)